cmd/kube-scheduler/app/testing/testserver.go

```diff
@@ -103,6 +103,9 @@ func StartTestServer(ctx context.Context, customFlags []string) (result TestServ
 		fs.AddFlagSet(f)
 	}
 	fs.Parse(customFlags)
+	if err := opts.ComponentGlobalsRegistry.Set(); err != nil {
```

Comment:
Isn't this already called inside `Validate`?

Reply:
Ideally we want `Set()` be called as early as possible, because there could be cases where FeatureGate is checked before `Validate`. The `Set` in `Validate` is a fallback in case `Set` is not already invoked.